### PR TITLE
[codex] Implement web admin app action catalog

### LINKED ISSUE #59: web admin: separate Kitu general and project pages in navigation and breadcrumbs
## 背景

web admin（backend/frontend）は、logic processor が保持するデータを諸元として扱い、アプリケーションごとに必要なページ・ビューを自由に構築していく使い方を想定する。

そのため、web admin のページ一覧と breadcrumb では、Kitu 共通の管理ビューとプロジェクト固有ビューを明確に分けたい。

## 現状

- `tools/kitu-web-admin/frontend/src/routes/+layout.svelte` の `nav` は `Overview`, `World`, `Logs` をフラットに表示している。
- SvelteKit の routes は以下のように `src/routes` 直下に置かれている。
  - `src/routes/+page.svelte`
  - `src/routes/world/+page.svelte`
  - `src/routes/logs/+page.svelte`
- breadcrumb 表示はまだない。

## 要件

- ページ一覧を `Kitu general` と `Project` に分けて表示する。
- サイドナビでは `Kitu general` と `Project` の間に区切り線を入れる。
- breadcrumb でも、現在のページが `Kitu general` 配下か `Project` 配下か分かるようにする。
- `Kitu general` には Kitu で共通となるページを置く。
  - 例: `Overview`, `World`, `Logs`, `Settings`
- `Project` にはプロジェクト固有のページを置く。
  - 例: `Level Designer`, `Story Sequencing`, `Game Parameters`
- SvelteKit のディレクトリ構造でルーティングが自動的に決まる形にできるなら、その形で `routes` を整理する。

## 提案するルーティング構成

SvelteKit の route groups を使い、URL には分類名を出さずに実装上の分類を持たせる案。

```text
tools/kitu-web-admin/frontend/src/routes/
  +layout.svelte
  (kitu-general)/
    +page.svelte              # /
    world/+page.svelte        # /world
    logs/+page.svelte         # /logs
    settings/+page.svelte     # /settings, 必要になった時点で追加
  (project)/
    level-designer/+page.svelte
    story-sequencing/+page.svelte
    game-parameters/+page.svelte
```

URL 上も分類を明示したい場合は、route groups ではなく `general/` と `project/` を実 URL に含める案もあり得る。ただし既存の `/`, `/world`, `/logs` を維持しやすいので、まずは route groups を優先する。

## 実装メモ

- `+layout.svelte` の `nav` を単一配列ではなく section 配列にする。
- section は最低限以下の情報を持つ。
  - `id`: `kitu-general` / `project`
  - `label`: `Kitu general` / `Project`
  - `items`: nav item list
- breadcrumb 生成にも同じ nav metadata を使い、サイドナビと breadcrumb の分類がずれないようにする。
- breadcrumb は header 内、Tick/Object 表示と競合しない位置に追加する。
- `Project` セクションは、まだページ実体がない場合でも metadata 側で追加しやすい構成にする。

## 受け入れ条件

- サイドナビで `Kitu general` と `Project` が区切り線付きで分かれて表示される。
- `Overview`, `World`, `Logs` は `Kitu general` に分類される。
- プロジェクト固有ページを追加するための `Project` セクションが用意されている。
- breadcrumb に `Kitu general > World`、`Kitu general > Logs` のような分類が表示される。
- SvelteKit の routes ディレクトリが、可能な限り分類に対応した構成になっている。
- 既存の `/`, `/world`, `/logs` が意図せず壊れない。
- frontend の `npm run check` が通る。

### LINKED ISSUE #60: web admin / kitu-cli: share app action definitions through the logic processor runtime
## 背景

web admin から実行できる操作は、同等の OSC イベント送出として `kitu-cli` からも実行できるようにしたい。

また、`enemy spawn`, `godmode`, `fast travel` のようなプロジェクト固有の抽象化された操作は、web admin の UI と `kitu-cli` のコマンド定義で共有したい。

一方で、`spawn object`, `move`, `reset` のような Kitu general な操作は framework 側の固定定義でよい。

## 重要な前提

プロジェクト固有の操作定義は、web admin backend の所有物ではなく、logic processor / application runtime backend 側の所有物とする。

つまり、`apps/<app-id>/kitu-app-actions.toml` のような定義ファイルは runtime が読み込み、runtime が action catalog と action execution capability を公開する。web admin backend はその source of truth ではなく、runtime から取得した catalog を web admin frontend に中継し、実行要求も runtime 側へ渡す役割に寄せる。

現在の demo web admin backend が runtime を同一プロセス内に持っている場合でも、それは vertical slice 上の配置であり、設計上の所有権は logic processor / application runtime 側に置く。

## 現状

- web admin frontend は `tools/kitu-web-admin/frontend/src/lib/osc-ir.ts` から WASM binding を呼び、OSC-IR JSON を作って WebSocket 送信している。
- backend は `tools/kitu-web-admin/backend/src/main.rs` で以下の admin OSC address を直接処理している。
  - `/admin/world/spawn`
  - `/admin/world/move`
  - `/admin/world/reset`
- `tools/kitu-cli/src/main.rs` は `kitu-cli not yet implemented` のスタブ。
- `crates/kitu-osc-ir` は OSC-IR の基礎型を持つが、操作 catalog や command schema はまだ持っていない。

## 解きたい問題

- web admin と `kitu-cli` で、同じ抽象操作が別々に定義されて乖離しないようにする。
- プロジェクトごとに追加される操作を、web admin のフォーム/ボタンと CLI から同じ catalog として参照できるようにする。
- 操作の最終出力は OSC-IR message/bundle として統一する。
- Kitu general な操作は framework 側で固定定義し、各 project が再定義しなくても使えるようにする。
- プロジェクト固有の操作定義は application runtime の一部として配布・ロードされるようにする。

### 1. 操作定義の共有 crate を追加する

例: `crates/kitu-app-actions`

この crate に、logic processor runtime、web admin backend、`kitu-cli` が共有する操作定義モデルを置く。名前は `admin` に限定せず、アプリケーション固有の runtime action catalog であることが分かるものにする。

```rust
pub struct AppActionDefinition {
    pub id: String,
    pub scope: AppActionScope, // KituGeneral | Project
    pub label: String,
    pub description: Option<String>,
    pub cli: CliCommandSpec,
    pub ui: UiActionSpec,
    pub inputs: Vec<ActionInputSpec>,
    pub output: OscTemplate,
}

pub enum AppActionScope {
    KituGeneral,
    Project { app_id: String },
}
```

責務:

- 操作 ID、表示名、scope、入力 schema、CLI command spec、UI rendering hint、OSC-IR template を表現する。
- 操作定義を `OscMessage` または `OscBundle` に materialize する。
- framework 固定の Kitu general 操作を提供する。
  - `spawn-object`
  - `move-object`
  - `reset-world`
- プロジェクト固有操作を runtime が追加ロードできる拡張点を持つ。

### 2. プロジェクト固有操作は runtime 側の manifest/schema で定義する

プロジェクト側には、操作 manifest を置けるようにする。

例:

```text
apps/<app-id>/kitu-app-actions.toml
# or
apps/<app-id>/kitu-app-actions.json
```

定義例:

```toml
[[actions]]
id = "enemy.spawn"
scope = "project"
label = "Enemy Spawn"

[actions.cli]
command = "enemy spawn"

[[actions.inputs]]
name = "enemy_type"
type = "string"
required = true

[[actions.inputs]]
name = "x"
type = "float"
required = true

[actions.output]
address = "/game/enemy/spawn"
args = ["$enemy_type", "$x"]
```

manifest は logic processor / application runtime がロードする。runtime は `kitu-app-actions` の parser / validator / OSC materializer を使い、読み込んだ catalog を外部へ公開する。

### 3. runtime が action catalog と execution capability を公開する

web admin backend ではなく、logic processor runtime が以下の能力を持つ。

- action catalog を返す。
- action id と入力値を受け取り、validation 後に OSC-IR を生成する。
- 生成した OSC-IR を runtime の既存 command/event pipeline に流す。

API 名は実装時に確定するが、概念としては以下。

- `GET /app-actions`
- `GET /app-actions/:id`
- `POST /app-actions/:id/run`

web admin backend は、必要に応じてこれらを proxy する。demo backend のように runtime を同一プロセスに持つ場合は、同じ crate API を直接呼ぶ。

### 4. CLI は runtime catalog からコマンドを構成する

`tools/kitu-cli` は runtime から app action catalog を取得し、以下を提供する。

- `kitu-cli app action list`
- `kitu-cli app action describe <action-id>`
- `kitu-cli app action run <action-id> --arg key=value ...`
- Kitu general 操作の sugar command
  - `kitu-cli world spawn --kind enemy --x 1 --y 0 --z 2`
  - `kitu-cli world move --id obj-1 --x 4 --y 0 --z 6`

動的な project command を完全に clap の nested subcommand に展開するか、まずは `run <action-id>` と key/value args で始めるかは実装コストを見て決める。初期実装は `run <action-id>` が堅い。

ローカル開発用途では CLI が manifest を直接読む fallback もあり得るが、source of truth は runtime が公開する catalog とする。

### 5. web admin は同じ runtime catalog を UI metadata として使う

web admin frontend は web admin backend 経由、または直接 runtime API 経由で app action catalog を取得する。

frontend は `UiActionSpec` と `ActionInputSpec` からフォームを生成し、submit 時は action run endpoint に投げる。

backend は catalog の所有者にならず、以下に集中する。

- runtime 接続設定
- web admin frontend 向け proxy / session 管理
- WebSocket event relay
- ログ、状態 snapshot、エラー表現の整形

### 6. web admin から kitu-cli を利用する、の解釈

web admin から OS process として `kitu-cli` を直接 spawn するより、まずは `kitu-cli` と web admin backend が同じ `kitu-app-actions` library API と runtime catalog を使う設計を優先する。

理由:

- web admin backend から CLI process を起動すると、認証、環境変数、path、ログ、権限、エラー表現が複雑になる。
- library API と runtime catalog を共有すれば、CLI と web admin が同じ validation / OSC materialization を使える。
- 必要になった場合だけ、backend に `kitu-cli` process adapter を追加できる。

推奨する境界:

```text
kitu-app-actions crate
  -> definition model
  -> manifest parser
  -> input validation
  -> OSC-IR generation

logic processor / application runtime
  -> loads project action manifests
  -> merges Kitu general actions + project actions
  -> exposes action catalog
  -> executes actions through runtime pipeline

kitu-cli
  -> CLI argument parsing
  -> runtime catalog lookup
  -> action execution request
  -> optional local manifest fallback for development

web admin backend
  -> runtime proxy/session layer
  -> calls runtime action catalog/run capability
  -> broadcasts state/logs

web admin frontend
  -> renders action catalog/forms
  -> calls backend or runtime action run API
```

## 実装ステップ案

1. `kitu-app-actions` crate を追加し、Kitu general action definitions を Rust 固定定義として実装する。
2. logic processor / application runtime に project action manifest loader を追加し、Kitu general actions と project actions を merge した catalog を持たせる。
3. runtime に action catalog/run capability を追加する。
4. `kitu-cli app action list/describe/run` を実装し、runtime catalog 経由で操作できるようにする。
5. web admin backend に runtime action catalog/run の proxy を追加する。
6. web admin frontend に action catalog 表示と汎用 action form を追加する。
7. `enemy.spawn` などをサンプル project action として定義する。
8. 既存の world spawn/move/reset UI を、直接 WASM builder 呼び出しから shared app action definition 経由に寄せる。

## 受け入れ条件

- Kitu general 操作の定義が logic processor runtime、web admin、`kitu-cli` で共有されている。
- プロジェクト固有操作の定義が `apps/<app-id>/kitu-app-actions.toml` など runtime 側の manifest に集約されている。
- runtime が Kitu general actions と project actions を merge した catalog を公開できる。
- `kitu-cli` から web admin と同等の OSC-IR message を送出できる。
- web admin backend は action definition の source of truth ではなく、runtime catalog/run capability を中継する。
- `enemy spawn`, `godmode`, `fast travel` のような project 操作を、CLI と web admin UI に二重定義せず追加できる見通しがある。
- Kitu general 操作と project 操作の scope が metadata 上で区別できる。
- `cargo test --workspace` の対象に共有 crate、runtime catalog、CLI のテストが含まれる。

### PR DESCRIPTION
**Pull request by Nagitch**
## Summary

- Split Web Admin navigation into `Kitu general` and `Project` sections using SvelteKit route groups while preserving `/`, `/world`, and `/logs` URLs.
- Added breadcrumbs backed by shared navigation metadata.
- Added `kitu-app-actions` for shared app action definitions, manifest loading, validation, and OSC-IR materialization.
- Wired the runtime, demo backend, frontend App Actions page, and `kitu-cli` to use the shared catalog.
- Added a demo project action manifest for `enemy.spawn`, `player.godmode`, and `map.fast-travel`.

Closes #59.
Closes #60.

## Validation

- `docker compose -f tools/kitu-web-admin/docker-compose.yml run --rm backend cargo fmt --all`
- `docker compose -f tools/kitu-web-admin/docker-compose.yml run --rm backend cargo test --workspace`
- `docker compose -f tools/kitu-web-admin/docker-compose.yml run --rm backend cargo clippy --all-targets --all-features -- -D warnings`
- `npm run check`
- `npm run lint`
- `docker exec -w /workspace/tools/kitu-web-admin/frontend kitu-web-admin-frontend-1 npm run build`
- `curl http://localhost:8787/health`
- `curl http://localhost:8787/app-actions`
- `curl -X POST http://localhost:8787/app-actions/spawn-object/run ...`
- Browser verification of `/app-actions` and `/world` breadcrumbs/navigation/forms at `http://localhost:5173`.

### CODE REVIEW

File: crates/kitu-app-actions/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,932 @@
...
+    }
+
+    /// Adds multiple action definitions.
+    pub fn extend_actions(
+        &mut self,
+        actions: impl IntoIterator<Item = AppActionDefinition>,
+    ) -> AppActionResult<()> {
+        for action in actions {
+            self.add_action(action)?;
+        }
```
[the remark this answers is not in the document]
Comment (Nagitch) [reply crates/kitu-app-actions/src/lib.rs]: Addressed in `3733cf7`. `extend_actions` now validates the full incoming batch against the existing catalog, including duplicate IDs, before mutating `self.actions`. I also added a regression test that verifies a failed duplicate batch leaves the catalog unchanged.

File: crates/kitu-app-actions/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,932 @@
...
+    Output,
+}
+
+fn parse_manifest_builders(source: &str) -> AppActionResult<Vec<ManifestActionBuilder>> {
+    let mut actions = Vec::new();
+    let mut current: Option<ManifestActionBuilder> = None;
+    let mut section = ManifestSection::Action;
+
+    for (line_index, raw_line) in source.lines().enumerate() {
+        let line = raw_line.split('#').next().unwrap_or("").trim();
```
[the remark this answers is not in the document]
Comment (Nagitch) [reply crates/kitu-app-actions/src/lib.rs]: Addressed in `3733cf7`. Comment stripping now tracks quoted strings and escapes, so `#` inside quoted TOML values is preserved while trailing comments still work. Added a regression test covering labels/descriptions/defaults/template literals containing `#`.
